### DISCUSSION TITLE
ZeroMQ-based srsUe and srsGnb is not able to decode Rach, RRC connected is not happening

### DISCUSSION THREAD

**Discussion by arpitmal** [srsran/srsRAN_Project#283] [General Help] [answered]
I am running with SRSRAN gnb latest code base 23.10.1 release, and trying to use ZeroMQ-based Setup steps mentioned in documentation, but UE is stuck in RACH phase.

sudo ./build/srsue/src/srsue ../ue_rf.conf
Active RF plugins: libsrsran_rf_uhd.so libsrsran_rf_zmq.so
Inactive RF plugins:
Reading configuration file ../ue_rf.conf...

Built in Release mode using commit fa56836b1 on branch master.

Opening 1 channels in RF device=zmq with args=tx_port=tcp://127.0.0.1:2001,rx_port=tcp://127.0.0.1:2000,base_srate=11.52e6
Supported RF device list: UHD zmq file
CHx base_srate=11.52e6
Current sample rate is 1.92 MHz with a base rate of 11.52 MHz (x6 decimation)
CH0 rx_port=tcp://127.0.0.1:2000
CH0 tx_port=tcp://127.0.0.1:2001
Current sample rate is 11.52 MHz with a base rate of 11.52 MHz (x1 decimation)
Current sample rate is 11.52 MHz with a base rate of 11.52 MHz (x1 decimation)
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Waiting PHY to initialize ... Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=331
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=361
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=391
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=421
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=451
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=481
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0xf, tti=511
^CStopping ..
q

--

If i try to run older version of gnb, i see RRC getting connected.
[gnb.log](https://github.com/srsran/srsRAN_Project/files/13218302/gnb.log)
[gnb_zmq.json](https://github.com/srsran/srsRAN_Project/files/13218304/gnb_zmq.json)
[ue.log](https://github.com/srsran/srsRAN_Project/files/13218305/ue.log)

**Comment by herlesupreeth** [comment, marked as answer]
Please refer to the answer here - https://github.com/srsran/srsRAN_Project/issues/263#issuecomment-1773756230

**Reply by arpitmal** [reply]
sudo ./build/srsue/src/srsue ../ue_rf.conf
Active RF plugins: libsrsran_rf_uhd.so libsrsran_rf_zmq.so
Inactive RF plugins:
Reading configuration file ../ue_rf.conf...

Built in Release mode using commit fa56836b1 on branch master.

Opening 1 channels in RF device=zmq with args=tx_port=tcp://127.0.0.1:2001,rx_port=tcp://127.0.0.1:2000,base_srate=11.52e6
Supported RF device list: UHD zmq file
CHx base_srate=11.52e6
Current sample rate is 1.92 MHz with a base rate of 11.52 MHz (x6 decimation)
CH0 rx_port=tcp://127.0.0.1:2000
CH0 tx_port=tcp://127.0.0.1:2001
Current sample rate is 11.52 MHz with a base rate of 11.52 MHz (x1 decimation)
Current sample rate is 11.52 MHz with a base rate of 11.52 MHz (x1 decimation)
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Waiting PHY to initialize ... Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
Warning: Failed to create thread with real-time priority. Creating it with normal priority: Operation not permitted
done!
Attaching UE...
Random Access Transmission: prach_occasion=0, preamble_index=0, ra-rnti=0x39, tti=334
Random Access Complete.     c-rnti=0x4601, ta=0
RRC Connected
Received RRC Release
--

after adding prach_config_index, it goes in connected , Ue receive RRC release immediately. 

    "cell_cfg": {
        "dl_arfcn": 368500,
        "band": 3,
        "channel_bandwidth_MHz": 10,
        "common_scs": 15,
        "plmn": "00101",
        "tac": 7,
        "prach": {
               "prach_config_index": 1
            },
        "pdcch": {
            "dedicated": {
                "ss2_type": "common",
                "dci_format_0_1_and_1_1": false
            }
        }
    },

**Reply by herlesupreeth** [reply]
Please check your APN configuration in 5GC

**Reply by arpitmal** [reply]
Thanks fixed core config and its working fine now.
